src/Symfony/Component/Validator/Exception/UnexpectedValueException.php

```diff
@@ -0,0 +1,32 @@
+<?php
+
+/*
+ * This file is part of the Symfony package.
+ *
+ * (c) Fabien Potencier <fabien@symfony.com>
+ *
+ * For the full copyright and license information, please view the LICENSE
+ * file that was distributed with this source code.
+ */
+
+namespace Symfony\Component\Validator\Exception;
+
+/**
+ * @author Christian Flothmann <christian.flothmann@sensiolabs.de>
+ */
+class UnexpectedValueException extends UnexpectedTypeException
+{
+    private $expectedType;
+
+    public function __construct($value, string $expectedType)
```

Comment:
what about `array $expectedTypes`. Right now `array or Traversable` is not really a friendly translation parameter. We should leverage the new message formatter here :)

Reply:
That assumes it's all unions. It will prevent consumers to use more complicated rules in future, e.g. intersection types. Also, I'm interested how would you improve message `array or Traversable`

Reply:
maybe

```
throw UnexpectedValueException::notUnionOf($value, $type, ...$type);
throw UnexpectedValueException::notIntersectionOf($value, $type, ...$type);
```